mass_mailing_list_dynamic/models/mail_mass_mailing_contact.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2017 Tecnativa - Jairo Llopis
+# Copyright 2019 Tecnativa - Victor M.M. Torres
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class MassMailingContact(models.Model):
+    _inherit = "mail.mass_mailing.contact"
+
+    @api.constrains("partner_id", "list_ids", "name", "email")
+    def _check_no_manual_edits_on_fully_synced_lists(self):
+        if self.env.context.get("syncing"):
+            return
+        for list_id in self.mapped('list_ids'):
```

Comment:
Here perhaps can use filtered to avoid do the "for loop"
```suggestion
        if any(self.mapped('list_ids').filtered(lambda x: x.dynamic and x.sync_method == "full")):
```